### DISCUSSION TITLE
File explorer color API

### DISCUSSION THREAD

**Discussion by kylejlin** [microsoft/vscode-discussions#71] [Extension Development QnA] [answered]
## Goal

I want to make an extension that allows users to configure the color of each file in the file explorer, using a `color_config.json` file at the root of the project.

## Example

If you write this in `color_config.json`...
```json
{
    "fileColors": [
        { "pattern": "src/tests/*", "color": "#0000ff" },
        { "pattern": "src/actions/*", "color": "#ff0000"}
    ]
}
```
...then the files in the `src/tests` directory will be blue, and the files in the `src/actions` directory will be red.

##

A very limited version of this already exists for Git--added files will show up as green, modified files as tan, etc. However, this only works for Git; as far as I know, there is no out-of-the-box way to configure colors in general.

So, I want to make an extension that lets users customize their file colors. However, I can't find any docs on color-related APIs. Do you guys know of any?

**Comment by r3m0t** [comment, marked as answer]
That's a file decoration

https://code.visualstudio.com/api/references/vscode-api#FileDecorationProvider

https://code.visualstudio.com/api/references/vscode-api#FileDecoration

**Comment by oreporan** [comment]
Check it out - https://marketplace.visualstudio.com/items?itemName=orepor.color-tabs-vscode-ext

**Reply by kylejlin** [reply]
Interesting. I want to make an extension to color the files in the _file explorer_ (not the active tab), in order to make it easier to visually locate files. However, maybe there is a good use case for your extension too. Nice work!
